### DISCUSSION TITLE
Need some help with the let's encrypt certificate

### DISCUSSION THREAD

**Discussion by thisisabi** [drakkan/sftpgo#1157] [Q&A]
Hello,

I am working on a requirement to have a SFTP server running on Google Compute which our clients can connect via a client of their choice. I was able to get it up and running and now I want to install a certificate to have the connection secure. I am following the documentation here - https://github.com/drakkan/sftpgo/blob/main/docs/howto/lets-encrypt-certificate.md#Obtaining-a-certificate-using-the-ACME-protocol-built-into-SFTPGo but it's not working.

Issues am facing - 
1. After I create the directory it wants and assign the permission, the command to "register the account and obtain certificates" is failing with some permission errors and I am not sure how to fix that.
2. I am not sure how to achieve this - "This directory must be publicly served by your web server.".

Could someone please help?

Abi

**Comment by drakkan** [comment]
> Hello,
> 
> I am working on a requirement to have a SFTP server running on Google Compute which our clients can connect via a client of their choice. I was able to get it up and running and now I want to install a certificate to have the connection secure. I am following the documentation here - https://github.com/drakkan/sftpgo/blob/main/docs/howto/lets-encrypt-certificate.md#Obtaining-a-certificate-using-the-ACME-protocol-built-into-SFTPGo but it's not working.
> 
> Issues am facing -
> 
> 1. After I create the directory it wants and assign the permission, the command to "register the account and obtain certificates" is failing with some permission errors and I am not sure how to fix that.

try something like `chown sftpgo:sftpgo <your dir>`, but this is probably not required for your use case, see later

> 2. I am not sure how to achieve this - "This directory must be publicly served by your web server.".

The guide makes the following assumption

"""
we have an existing web server already running on port 80 for the sftpgo.com domain and the web root path is /var/www/sftpgo.com
"""

if you don't have an existing web server you don't need to set a web root path, SFTPGo can solve the `HTTP-01` challenge by itself
> 
> Could someone please help?
> 
> Abi

**Comment by coffseducation** [comment]
I deleted the website.
Re-run acme  - Did not work.

Issued command: fuser -k 80/tcp

Re-run acme
![image](https://github.com/drakkan/sftpgo/assets/99161534/0d1bad99-1a6c-4f60-a21e-84657397e093)

but now sftpgo does not load at all!  😐

edit: Still works with http

**Comment by coffseducation** [comment]
root@backup:/etc/sftpgo# ls
env.d  id_ecdsa  id_ecdsa.pub  id_ed25519  id_ed25519.pub  id_rsa  id_rsa.pub  sftpgo.json

root@backup:/var/lib/sftpgo/certs# ls
acme-v02.api.letsencrypt.org         backup.mysite.academy.json
backup.mysite.academy.crt  backup.mysite.academy.key


sftpgo serve
2024-01-20T10:31:12.927 INF telemetry server not started, disabled in config file
2024-01-20T10:31:12.928 INF No host keys configured and "id_rsa" does not exist; try to create a new host key
2024-01-20T10:31:12.951 ERR could not start HTTP server: listen tcp :8080: bind: address already in use

**Comment by drakkan** [comment]
> root@backup:/etc/sftpgo# ls env.d id_ecdsa id_ecdsa.pub id_ed25519 id_ed25519.pub id_rsa id_rsa.pub sftpgo.json
> 
> root@backup:/var/lib/sftpgo/certs# ls acme-v02.api.letsencrypt.org backup.mysite.academy.json backup.mysite.academy.crt backup.mysite.academy.key
> 
> sftpgo serve 2024-01-20T10:31:12.927 INF telemetry server not started, disabled in config file 2024-01-20T10:31:12.928 INF No host keys configured and "id_rsa" does not exist; try to create a new host key 2024-01-20T10:31:12.951 ERR could not start HTTP server: listen tcp :8080: bind: address already in use

![Schermata del 2024-01-20 11-34-40](https://github.com/drakkan/sftpgo/assets/553263/7697e9d8-ef60-43da-8112-743c6b918017)

```
sudo systemctl restart sftpgo
```

**Reply by coffseducation** [reply]
Thanks I was looking for such a command and could only find 'sftpgo serve'

Still not working on https. 😔

**Comment by drakkan** [comment]
> Thanks I was looking for such a command and could only find 'sftpgo serve'
> 
> Still not working on https. 😔

`https://<domain name>:8080/` should work now if you have managed to obtain a certificate and have selected HTTP as the protocol to use the certificate obtained from the WebUI

**Reply by coffseducation** [reply]
Oh! I may not have selected a protocol. 😁

**Comment by coffseducation** [comment]
🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉
THANK YOU!!!!
🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉
WORKING!!!!!!!
